### PR TITLE
EN-2312 EN-2353 DE Component installation improvements

### PR DESCRIPTION
**Pull request by zonia3000**
Hi @joewhite101, could you please have a look at this PR, I think it is the most critical one, in particular the logic contained in the [CommandExecutor](https://github.com/entando/entando-components/blob/1e496f89d39bd87bafdd8d96078f3d227a803a8d/plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/install/CommandExecutor.java) class.

I thought this as a replacement for the [SelfRestCaller](https://github.com/entando/entando-core/blob/master/engine/src/main/java/org/entando/entando/aps/system/init/SelfRestCaller.java), which is currently used during the installation and uses the [ResponseBuilder](https://github.com/entando/entando-core/blob/master/engine/src/main/java/org/entando/entando/aps/system/services/api/server/ResponseBuilder.java), that uses the reflection for calling the legacy controllers and inserting for example content models and content types. The general idea is to exploit the SpEL to provide a flexible mechanism to call an arbitrary class, even one that is available only at runtime (this is the case of the CMS controllers, that are not a dependency of the DE plugin itself, and can be useful to support future component types).
I also thought about providing some "template operations", for the most common use cases (e.g. adding a content model), in order to simplify the construction of the component.xml file, which describes the installation needs.

Basically we can have in the component.xml a sequence of operations, defined like this:
```xml
<execution>
    <!-- template operation -->
    <op>#call('createContentType', 'demo_1/data/demo_content_type.json')</op>
    <!-- custom operation -->
    <op>@myBean.doSomething()</op>
</execution>
```

This can be used for the uninstallation too.

Other changes in this PR:
- support for static resources (simply based on the `IStorageManager` and some directory structure conventions)
- additional auditing columns in the job table
- changed the installation API using the component id instead of the job id, as required by FE (API returns the last job for a given component).

### CODE REVIEW

**Comment by entando-service** [comment]
All is Well

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/install/CommandExecutor.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+        try {
+            context.registerFunction("bindingResult", CommandUtils.class.getDeclaredMethod("getBindingResult"));
+            context.registerFunction("call", Command.class.getDeclaredMethod("newInstance", String.class, String[].class));
+        } catch (NoSuchMethodException ex) {
+            throw new BeanInitializationException("Unable to register SpEL functions", ex);
+        }
+
+        try (InputStream in = getClass().getClassLoader().getResourceAsStream("component/plugins/digitalExchange/commands.xml")) {
+            commands = JAXB.unmarshal(in, Commands.class);
```
Comment (rzsouza) [inline plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/install/CommandExecutor.java]: In this line the input stream (in) can be null. Maybe treat it or log something

Comment (zonia3000) [reply plugins/entando-plugin-jpdigital-exchange/src/main/java/org/entando/entando/aps/system/services/digitalexchange/install/CommandExecutor.java]: Done

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by zonia3000** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by zonia3000** [comment]
retest this please

**Comment by entando-service** [comment]
Something went wrong. Investigate!

**Comment by zonia3000** [comment]
retest this please

**Pull request merged**